tools/Az.Tools.Predictor/Az.Tools.Predictor/AzContext.cs

```diff
@@ -28,11 +29,46 @@ namespace Microsoft.Azure.PowerShell.Tools.AzPredictor
     /// <summary>
     /// The class for the current Azure PowerShell context.
     /// </summary>
-    internal sealed class AzContext : IAzContext
+    internal sealed class AzContext : IAzContext, IDisposable
     {
         private const string InternalUserSuffix = "@microsoft.com";
         private static readonly Version DefaultVersion = new Version("0.0.0.0");
 
+        private PowerShell _powerShellRuntime;
+        private PowerShell PowerShellRuntime
+        {
+            get
+            {
+                if (_powerShellRuntime == null)
+                {
+                    _powerShellRuntime = PowerShell.Create(DefaultRunspace);
+                }
+
+                return _powerShellRuntime;
+            }
+        }
+
+        private readonly Lazy<Runspace> _defaultRunspace = new(() =>
+                {
+                    // Create a mini runspace by remove the types and formats
+                    InitialSessionState minimalState = InitialSessionState.CreateDefault2();
+                    minimalState.Types.Clear();
+                    minimalState.Formats.Clear();
+                    // Refer to the remarks for the property DefaultRunspace.
+                    minimalState.ImportPSModule("Az.Accounts", "Az.Storage", "Az.Network", "Az.KeyVault", "Az.Compute", "Az");
```

Comment:
@dingmeng-xue What do you if I load the 'Az' module ahead of time? The purpose of this is to save the time in `GetSuggestion` to parse positional parameters and return the predictions.